server/routes.js

```diff
@@ -548,3 +548,34 @@ exports.mountQuintypeAt = function(app, mountAt) {
     }
   });
 };
+
+/**
+ * ampRoutes uses quintype-amp library to handle amp pages
+ * currently, only story pages with default styling are supported
+ */
+exports.ampRoutes = function(app, { opts } = {}) {
+  const { Story } = require("./api-client");
+  const { ampifyStory } = require("@quintype/amp");
+  getWithConfig(
+    app,
+    "/amp/story/*",
+    async (req, res, next, { client, config }) => {
+      try {
+        const slug = req.path.replace("^/amp/story", "");
+        const ampConfig = await client.getAmpConfig();
+        const story = await Story.getStoryBySlug(client, slug);
+        const ampHtml = ampifyStory({
+          story,
+          config: { ...ampConfig, ...config },
+          client,
+          opts
+        });
+
+        if (ampHtml instanceof Error) throw ampHtml;
+        story ? res.send(ampHtml) : next();
+      } catch (e) {
```

Comment:
don't use exeptions for control flow. If this is an exception, return 500. If story is null, return next()

Reply:
calling next(err) will return a 500 [https://expressjs.com/en/guide/error-handling.html](https://expressjs.com/en/guide/error-handling.html) 
Should I `res.send(500)` instead in the catch block?

Reply:
I did not know that. This os fine